pkgs/by-name/et/etherpad-lite/package.nix

```
@@ -0,0 +1,87 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  fetchpatch,
+  gitUpdater,
+  pnpm_9,
+  makeWrapper,
+  nodejs,
+}:
+
+let
+  pnpm = pnpm_9;
+in
+stdenv.mkDerivation (finalAttrs: {
+  pname = "etherpad-lite";
+  version = "2.3.0";
+
+  src = fetchFromGitHub {
+    owner = "ether";
+    repo = "etherpad-lite";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-OomZ1oziEGgLJfYyfkHZOPRHfrrWc4XLAsZA4cn0btA=";
+  };
+
+  patches = [
+    ./outdir.patch
+    # etherpad expects to read and write $out/lib/var/installed_plugins.json
+    # FIXME: this patch disables plugin support
+    ./dont-fail-on-plugins-json.patch
+  ];
+
+  pnpmDeps = pnpm.fetchDeps {
+    inherit (finalAttrs) pname version src;
+    hash = "sha256-nhiPopGLCeCHiqEQ3solwuLwkDnHTH3otbxIJmbuQAA=";
+  };
+
+  nativeBuildInputs = [
+    pnpm.configHook
+    makeWrapper
+  ];
+
+  buildInputs = [
+    nodejs
+  ];
+
+  buildPhase = ''
+    runHook preBuild
+    NODE_ENV="production" pnpm run build:etherpad
+    runHook postBuild
+  '';
+
+  # Upstream scripts uses `pnpm run prod` which is equivalent to
+  # `cross-env NODE_ENV=production node --require tsx/cjs node/server.ts`
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/{lib,bin}
+    cp -r node_modules ui src doc admin $out/lib
+    makeWrapper ${lib.getExe nodejs} $out/bin/etherpad-lite \
+      --inherit-argv0 \
+      --add-flags "--require tsx/cjs $out/lib/node_modules/ep_etherpad-lite/node/server.ts" \
+      --suffix PATH : "${lib.makeBinPath [ pnpm ]}" \
+      --set NODE_PATH "$out/lib/node_modules:$out/lib/node_modules/ep_etherpad-lite/node_modules" \
+      --set-default NODE_ENV production
+    find $out/lib -xtype l -delete
+    runHook postInstall
+  '';
+
+  passthru.updateScript = gitUpdater { };
```

This will also update `pnpmDeps.hash`
```suggestion
  passthru.updateScript = nix-update-script { };
```